hudi-client/hudi-java-client/src/main/java/org/apache/hudi/table/action/commit/BaseJavaCommitActionExecutor.java

```diff
@@ -132,6 +132,11 @@ protected void updateIndex(List<WriteStatus> writeStatuses, HoodieWriteMetadata<
     result.setWriteStatuses(statuses);
   }
 
+  protected void updateIndexAndCommitIfNeeded(List<WriteStatus> writeStatuses, HoodieWriteMetadata<List<WriteStatus>> result) {
```

Comment:
@caidezhi Sorry for the delay, there is a PR https://github.com/apache/hudi/pull/2382 to support COW in Java client which includes the changes here.

Reply:
@leesf if it is already covered by other PR, please feel free to close this one. Thanks for your review.